src/import.rs

```diff
@@ -272,6 +274,50 @@ impl ImportObjectBuilder {
 
         Ok(ImportObject(sys::ImportObject::Import(inner)))
     }
+
+    /// Creates a new [ImportObject] with host data.
+    ///
+    /// # Argument
+    ///
+    /// * `name` - The name of the [ImportObject] to create.
+    ///
+    /// * `host_data` - The host data to be stored in the module instance.
+    ///
+    /// * `finalizer` - the function to drop the host data.
+    ///
+    /// # Error
+    ///
+    /// If fail to create the [ImportObject], then an error is returned.
+    pub fn build_with_data<T>(
+        self,
+        name: impl AsRef<str>,
+        host_data: &mut T,
+        finalizer: Option<Finalizer>,
+    ) -> WasmEdgeResult<ImportObject> {
```

Comment:
This function creates an `ImportObject` that does not have a dependency on `&mut host_data` in terms of its lifetime. This could potentially lead to memory safety issues.

Reply:
Accepted. Thanks a lot!